sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultVolumeSpec.java

```diff
@@ -86,15 +105,33 @@ public int hashCode() {
 
     @Override
     public String toString() {
-        return String.format("%s, type: '%s', container-path: '%s'",
-                super.toString(),
-                getType(),
-                getContainerPath());
+        return ToStringBuilder.reflectionToString(this);
     }
 
     private static Protos.Value scalarValue(double value) {
         Protos.Value.Builder builder = Protos.Value.newBuilder().setType(Protos.Value.Type.SCALAR);
         builder.getScalarBuilder().setValue(value);
         return builder.build();
     }
+
+    private void validateVolume() {
+        validateResource();
+        ValidationUtils.matchesRegex(this, "containerPath", containerPath, VALID_CONTAINER_PATH_PATTERN);
+
+        if (type == Type.MOUNT) {
+            ValidationUtils.nonEmptyAllowNull(this, "profiles", profiles);
```

Comment:
`nonEmpty()`

Reply:
If we go with empty lists, this should be simply `nonNull()` instead since we do support MOUNT volumes with no profile.